src/simulations/set_generator.py

```diff
@@ -51,6 +53,57 @@ def __iter__(self):
     for _ in range(self.num_sets):
       yield self.set_size
 
+def _choice_fast(n, m, random_state=np.random.RandomState()):
+    """Chooses m numbers or objects from population n.
+
+    O(m) space-optimal algorithm for generating m random indices for list 
```

Comment:
Can you put a reference of some kind for the algorithm into the comment?

Reply:
Resolved in 38fadaa